### PR TITLE
Add int8 support in fused_multi_transformer_pass and fuse_multi_transformer_layer_pass

### PR DESCRIPTION
**Pull request by RichardWooSJTU**
### PR types

Others
### PR changes

Others
### Describe

PR https://github.com/PaddlePaddle/Paddle/pull/45907 and PR https://github.com/PaddlePaddle/Paddle/pull/47541 finshed fusing fp16/32 GPT3 model to a single fused_multi_transformer op, which can speed up inference of GPT3. In this PR, we add int8 support, the operations added is as follows:

1. Add delete_weight_quant_dequant_linear_op_encoder/decoder_pass: For models with new quantization format, quant/dequant_linear op will be used to fakely quantize weight, this pass can delete these nodes and move quantization info to nodes following them:
image
==》
image
2. Add enable_int8 attr of graph to distinguish int8 op pattern from  fp16/32 pattern in fused_multi_transformer_encoder/decoer pass
3. Add calculation of input scale and output scale and transpose operation of int8 weight (which is needed by cublasLt imma layout)
4. Add some optimization of fused_multi_transformer_int8 op

### CODE REVIEW

File: paddle/fluid/operators/fused/cublaslt.h [hunk 1]
```diff
@@ -176,13 +208,14 @@ class CublasLtHelper {
...
-#if __CUDA_ARCH__ >= 800 && CUDA_VERSION >= 11020
-                                 &algo,
+#if CUDA_VERSION >= 11020
+                                 &algo_,
+                                 //  nullptr,// workspace
+                                 workspace,
+                                 workspace_size_,
 #else
-                                 nullptr,
+                                 hshshsh
```
Comment (MARD1NO) [inline paddle/fluid/operators/fused/cublaslt.h]: 这里是不是改错了

File: paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h [hunk 2]
```diff
@@ -543,8 +541,10 @@ __global__ __launch_bounds__(THREADS_PER_CTA) void fused_fast_ln_fwd_kernel(
           // dropout(x) + residual
           if (std::is_same<InType, int32_t>::value) {
             T tmp = (static_cast<T>(static_cast<float>(x_input[it][jt]) *
-                                    quant_last_in_scale /
-                                    dequant_out_scale[it][jt]) +
+                                    dequant_out_scale[it][jt]
+                                    // quant_last_in_scale /
```
Comment (MARD1NO) [inline paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h]: 要不直接删了

File: paddle/fluid/operators/fused/cublaslt.h (hunk 1 above)
Comment (RichardWooSJTU) [reply paddle/fluid/operators/fused/cublaslt.h]: 收到 测试代码 忘记删除了

File: paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h (hunk 2 above)
Comment (RichardWooSJTU) [reply paddle/fluid/operators/fused/fused_layernorm_residual_dropout_bias.h]: done~

**Review by chenwhql — APPROVED** [review]
麻烦再提交一个PR完善报错信息

**Review by heavengate — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]
But need to unify delete_weight_dequant_linear_op pass

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**